backend-rust/src/indexer.rs

```diff
@@ -1340,3 +1380,134 @@ impl PreparedBakerEvent {
         Ok(())
     }
 }
+
+struct PreparedModuleDeployed {
+    block_height: i64,
+    deployment_transaction_index: i64,
+    module_reference: String,
+    schema: Option<String>,
+}
+
+impl PreparedModuleDeployed {
+    async fn prepare(
```

Comment:
Why have the prepare and save functions separate? You can't do anything with this struct other than call save afterwards anyway it seems like.